### PR TITLE
Build: Remove old maven deploy support

### LINKED ISSUE #20375: Building 5.0.0 SNAPSHOTs and using TransportClient in other projects in hard
I'm trying to build a SNAPSHOT version of the latest master to be able to integrate it in a Test project.

After cloning the elasticsearch repo, I'm running:

```sh
gradle install
```

It works well.

Then I'm creating another project (https://github.com/dadoonet/legacy-search) where I want to use this SNAPSHOT version I just built.
In my  `pom.xml`, I'm writing:

```xml
<dependency>
    <groupId>org.elasticsearch.client</groupId>
    <artifactId>transport</artifactId>
    <version>5.0.0-alpha6-SNAPSHOT</version>
</dependency>
```

But sadly, some artifacts are not available in the standard `.m2` dir. When I run `mvn install` I'm getting:

```
[WARNING] The POM for org.elasticsearch.plugin:transport-netty3-client:jar:5.0.0-alpha6-SNAPSHOT is missing, no dependency information available
[WARNING] The POM for org.elasticsearch.plugin:reindex-client:jar:5.0.0-alpha6-SNAPSHOT is missing, no dependency information available
[WARNING] The POM for org.elasticsearch.plugin:lang-mustache-client:jar:5.0.0-alpha6-SNAPSHOT is missing, no dependency information available
[WARNING] The POM for org.elasticsearch.plugin:transport-netty4-client:jar:5.0.0-alpha6-SNAPSHOT is missing, no dependency information available
[WARNING] The POM for org.elasticsearch.plugin:percolator-client:jar:5.0.0-alpha6-SNAPSHOT is missing, no dependency information available
```

So basically the `pom.xml` for `org.elasticsearch.client:transport` and `org.elasticsearch:elasticsearch` have been published in the local `.m2` but I'm missing other jars/pom.

For example:

```
$ ls -l ~/.m2/repository/org/elasticsearch/plugin/transport-netty3-client
total 0
drwxr-xr-x  9 dpilato  staff  306 Sep  7 17:28 5.0.0-alpha5
```

We only have "released" versions here.

Is this something we can fix or does it mean we can't really test the transport client until a release or a snapshot has been published on elastic repository or on sonatype?

I believe the same should happen with Gradle unless there is a workaround?

### PR DESCRIPTION
**Pull request by rjernst**
This change removes the old maven deploy that we have in parallel to
maven-publish, and makes maven-publish fully work with publishing to
maven local. Using `gradle publishToMavenLocal` should be used to
publish to .m2.

Note that there is an unfortunate hack that means for
zip artifacts we must first create/publish a dummy pom file, and then
follow that with the real pom file. It would be nice to have the pom
file contains packaging=zip, but maven central then requires sources and
javadocs. But our zips are really just attached artifacts, so we already
set the packaging type to pom for our zip files. This change just works
around a limitation of the underlying maven publishing library which
silently skips attached artifacts when the packaging type is set to pom.

relates #20164
closes #20375

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/ingest/PipelineStore.java [hunk 1]
```diff
@@ -45,7 +45,7 @@
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.xcontent.XContentHelper;
 
-public class PipelineStore extends AbstractComponent implements ClusterStateListener {
+public class    PipelineStore extends AbstractComponent implements ClusterStateListener {
```
Comment (jasontedor) [inline core/src/main/java/org/elasticsearch/ingest/PipelineStore.java]: This change is probably accidental and I think should be removed.

Comment (rjernst) [inline core/src/main/java/org/elasticsearch/ingest/PipelineStore.java]: Oops, no idea how that got there. Removed in 4732a23

**Comment by rjernst** [comment]
@jasontedor any more thoughts on this?

**Comment by rjernst** [comment]
Ping for someone to look at this.

**Comment by nik9000** [comment]
> Ping for someone to look at this.

Sorry, sorry, sure. I'll look. Is this just for 5.0 or for all the 5+ branches?

**Comment by rjernst** [comment]
@nik9000 It is for all the branches: I updated the labels.

**Review by nik9000 — APPROVED** [review]
Code changes make sense - it slices out a good pile of funky maven juggling.

I pulled it locally and played with it and it worked. I spot checked a few of the poms and they seemed sane to me.

**Pull request merged**